### PR TITLE
feature (ch05_p01): add easy-to-understand solution

### PR DESCRIPTION
**Pull request by ryoyama0508**
Added a solution `chapter_05/p01_insertion.py` since the original solution is a bit hard to read and understand for me. I hope this solution will be helpful for people in the same situation.

By the way, I personally think we should change the test cases because the question in the book says _"You are given two 32-bit __numbers__, N and M, and two bit positions, i and j. (question continues)"_. So I think the input of the test case should be __numbers(int)__, not string. My understanding might be weird or wrong since I'm not native speaker.

### CODE REVIEW

**Comment by ryoyama0508** [comment]
I changed my code since flake8 doesn't let me use `.format`.

File: chapter_05/p01_insertion.py [hunk 1]
```diff
@@ -25,9 +25,26 @@ def bits_insertion(n, m, i, j):
...
+    # change str into int in order to do bit manipulation
+    int_n, int_m = int(n, 2), int(m, 2)
+
+    # do a liner search through the bits of M (tail to head)
+    # and if you find 1, do a bit insertion to N
+    for idx in range(j - i + 1):
+        if (int_m >> idx) & 1 != 0:
+            int_n |= 1 << (idx + i)
+
+    return bin(int_n)[2:]
```
Comment (brycedrennan) [inline chapter_05/p01_insertion.py:38]: f-strings are the replacement for `.format` 

So you could do:
```python
 return f"{int_n:b}"
```

File: chapter_05/p01_insertion.py [hunk 2]
```diff
@@ -25,9 +25,26 @@ def bits_insertion(n, m, i, j):
     return bit_str.zfill(num_bits)  # pad string to num_bits
 
 
+def bits_insertion_easy_to_understand(n, m, i, j):
+    # change str into int in order to do bit manipulation
```
Comment (brycedrennan) [inline chapter_05/p01_insertion.py:29]: I agree that this should just be starting with integers.

**Comment by brycedrennan** [comment]
Thanks! I just had some minor suggestions

**Comment by ryoyama0508** [comment]
Thanks for the suggestion. Changed the output to f-string. And also changed the input from string to integers. I wasn't sure what should I do to the other solution, so I just left it as it is.

**Review by brycedrennan — CHANGES_REQUESTED** [review]
Thanks for those updates. 

It's possible I'm mistaken but it seems this solution does not work if n is set to `int("11111111111", 2)` - the function outputs `11111111111`

Let me know what you think

**Comment by ryoyama0508** [comment]
> It's possible I'm mistaken but it seems this solution does not work if n is set to `int("11111111111", 2)` - the function outputs `11111111111`
> 
> Let me know what you think

I think you're completely right. I fixed that part.

Also, thanks for cleaning the test cases! Appreciate it.

**Comment by brycedrennan** [comment]
Thanks - looks good!

**Pull request merged**